carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/core/capability/AbstractCapabilities.java

```diff
@@ -112,11 +112,20 @@ protected DesiredCapabilities initCapabilities(DesiredCapabilities capabilities)
             capabilities = addChromeOptions(capabilities);
         }
 
-        if (Configuration.getBoolean(Parameter.HEADLESS)
-                && !BrowserType.FIREFOX.equalsIgnoreCase(browser) && !BrowserType.CHROME.equalsIgnoreCase(browser)) {
-            LOGGER.error(String.format("Headless mode isn't supported by %s browser", browser));
+        if (Configuration.getBoolean(Parameter.HEADLESS)) {
+            if (BrowserType.FIREFOX.equalsIgnoreCase(browser)
+                    || BrowserType.CHROME.equalsIgnoreCase(browser)
+                    && Configuration.getDriverType().equalsIgnoreCase(SpecialKeywords.DESKTOP)) {
+                LOGGER.info("Browser will be started in headless mode. VNC and Video will be disabled.");
+                R.CONFIG.put("capabilities.enableVNC", "false");
```

Comment:
don't override capabilities.enableVNC via R.CONFIG.put. This headless arg might present only in this thread for this particular browsersession but you disable for all global test run context!
putting valid args to capabilities is enough